integration/helpers.go

```diff
@@ -456,3 +456,34 @@ func createNestedDocument(n int, arr bool) any {
 
 	return bson.D{{"v", child}}
 }
+
+// ResultPushdown stores the information about expected pushdown results for a single or multiple backends.
+// For example if both pg and SQlite backends are expected to pushdown, the `PgPushdown + SQLitePushdown` operation
+// can be used.
+type ResultPushdown uint8
+
+const (
+	NoPushdown     ResultPushdown = 0
+	PgPushdown     ResultPushdown = 1
+	SQLitePushdown ResultPushdown = 2
+
+	AllPushdown ResultPushdown = PgPushdown + SQLitePushdown
```

Comment:
I think we want to think about how easy to adjust this when we add new backend with no pushdown. I like the introduction of current number enum thing, having constant 0, 1, 2, 4, 8 etc will allow us to do addition without collision :hugs: . By the way, naming `AllPushdown` would require modifying this name when a new backend is added or hana is using this test. I think it can be named this specifically to indicate it's pg and sqlite pushdown, something like `PgSQLitePushdown ResultPushdown = PgPushdown + SQLitePushdown`? :thinking:

Reply:
Good concern, I'm not quiet sure about `PgSQLitePushdown` idea though. My goal with this approach was to allow syntax like `PgPushdown+SQLitePushdown` to make it readable by just looking at the test case. What if we introduce another backend (`hana` for example), to stay consistent we would need to Introduce `HanaPgPushdown` `SQliteHanaPushdown`, etc...

I have 2 ideas though:
- we can just leave a comment above `AllPushdown` that it should be updated with each new backend
- we could assign The `0xFF` value to the AllPushdown, so it will be able to match every new backend introduced

WDYT?

Reply:
Great idea! I like how it's done :hugs: